pkg/controllers/job/job_controller.go

```diff
@@ -151,39 +154,43 @@ func (cc *jobcontroller) Initialize(opt *framework.ControllerOption) error {
 
 	factory := informerfactory.NewSharedInformerFactory(cc.vcClient, 0)
 	cc.vcInformerFactory = factory
-	cc.jobInformer = factory.Batch().V1alpha1().Jobs()
-	cc.jobInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    cc.addJob,
-		UpdateFunc: cc.updateJob,
-		DeleteFunc: cc.deleteJob,
-	})
-	cc.jobLister = cc.jobInformer.Lister()
-	cc.jobSynced = cc.jobInformer.Informer().HasSynced
-
-	cc.cmdInformer = factory.Bus().V1alpha1().Commands()
-	cc.cmdInformer.Informer().AddEventHandler(
-		cache.FilteringResourceEventHandler{
-			FilterFunc: func(obj interface{}) bool {
-				switch v := obj.(type) {
-				case *busv1alpha1.Command:
-					if v.TargetObject != nil &&
-						v.TargetObject.APIVersion == batchv1alpha1.SchemeGroupVersion.String() &&
-						v.TargetObject.Kind == "Job" {
-						return true
-					}
+	if utilfeature.DefaultFeatureGate.Enabled(features.WorkLoadSupport) {
+		cc.jobInformer = factory.Batch().V1alpha1().Jobs()
+		cc.jobInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc:    cc.addJob,
+			UpdateFunc: cc.updateJob,
+			DeleteFunc: cc.deleteJob,
+		})
+		cc.jobLister = cc.jobInformer.Lister()
+		cc.jobSynced = cc.jobInformer.Informer().HasSynced
+	}
 
-					return false
-				default:
-					return false
-				}
-			},
-			Handler: cache.ResourceEventHandlerFuncs{
-				AddFunc: cc.addCommand,
+	if utilfeature.DefaultFeatureGate.Enabled(features.QueueCommandSync) {
+		cc.cmdInformer = factory.Bus().V1alpha1().Commands()
+		cc.cmdInformer.Informer().AddEventHandler(
+			cache.FilteringResourceEventHandler{
+				FilterFunc: func(obj interface{}) bool {
+					switch v := obj.(type) {
+					case *busv1alpha1.Command:
+						if v.TargetObject != nil &&
```

Comment:
Suggest to be simpled as ` return v.TargetObject != nil && v.TargetObject.APIVersion == batchv1alpha1.SchemeGroupVersion.String() &&
							v.TargetObject.Kind == "Job" `